plonky2/src/gadgets/arithmetic.rs

```diff
@@ -189,9 +189,17 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
 
     /// Computes `x + y`.
     pub fn add(&mut self, x: Target, y: Target) -> Target {
-        let one = self.one();
+        let res = self.arithmetic(F::ONE, F::ONE, x, x, y);
```

Comment:
This is `1 * x * x + 1 * y`, but it should be `1 * x * 1 + 1 * y`. Same with the subtract. Check out related implementation in #10